rero_ils/modules/documents/listener.py

```diff
@@ -39,158 +39,178 @@ def enrich_document_data(sender, json=None, record=None, index=None,
     :param index: The index in which the record will be indexed.
     :param doc_type: The doc_type for the record.
     """
-    if index.split('-')[0] == DocumentsSearch.Meta.index:
-        # HOLDINGS
-        holdings = []
-        document_pid = record['pid']
-        es_holdings = HoldingsSearch()\
-            .filter('term', document__pid=document_pid)\
+    if index.split('-')[0] != DocumentsSearch.Meta.index:
```

Comment:
Maybe this function is too long and can be splited into several sub functions.

Reply:
Sure !
The "sourcery" plugin has a very poor performance indicator for this function :'(
But refactoring this function is an huge work. Maybe in a next PR :-)